{"metadata":{"title":"getRetryWaitTime(error:attempt:)","required":true,"role":"symbol","modules":[{"name":"SotoCore"}],"externalID":"s:8SotoCore11RetryPolicyP03getC8WaitTime5error7attemptAA0C6StatusOSgs5Error_p_SitF","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"getRetryWaitTime"},{"kind":"text","text":"("},{"text":"error","kind":"externalParam"},{"kind":"text","text":": "},{"text":"Error","preciseIdentifier":"s:s5ErrorP","kind":"typeIdentifier"},{"kind":"text","text":", "},{"text":"attempt","kind":"externalParam"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Int","preciseIdentifier":"s:Si"},{"kind":"text","text":") -> "},{"text":"RetryStatus","kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore11RetryStatusO"},{"text":"?","kind":"text"}],"roleHeading":"Instance Method","symbolKind":"method"},"schemaVersion":{"minor":3,"major":0,"patch":0},"abstract":[{"type":"text","text":"Returns whether we should retry and how long we should wait before retrying"}],"sections":[],"kind":"symbol","primaryContentSections":[{"declarations":[{"tokens":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"getRetryWaitTime"},{"text":"(","kind":"text"},{"text":"error","kind":"externalParam"},{"kind":"text","text":": "},{"text":"Error","preciseIdentifier":"s:s5ErrorP","kind":"typeIdentifier"},{"kind":"text","text":", "},{"kind":"externalParam","text":"attempt"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Int","preciseIdentifier":"s:Si"},{"kind":"text","text":") -> "},{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus","text":"RetryStatus","kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore11RetryStatusO"},{"text":"?","kind":"text"}],"languages":["swift"],"platforms":["macOS"]}],"kind":"declarations"},{"kind":"parameters","parameters":[{"name":"error","content":[{"inlineContent":[{"type":"text","text":"Error returned by HTTP client"}],"type":"paragraph"}]},{"name":"attempt","content":[{"inlineContent":[{"type":"text","text":"Retry attempt number"}],"type":"paragraph"}]}]}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)"},"hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy"]]},"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)"]}],"references":{"doc://codes.soto.soto-core/documentation/SotoCore/RetryStatus":{"title":"RetryStatus","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus","navigatorTitle":[{"kind":"identifier","text":"RetryStatus"}],"role":"symbol","kind":"symbol","abstract":[{"text":"Return value for ","type":"text"},{"type":"codeVoice","code":"RetryPolicy.getRetryWaitTime"},{"text":". Either retry after time amount or donâ€™t retry","type":"text"}],"fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"RetryStatus"}],"type":"topic","url":"\/documentation\/sotocore\/retrystatus"},"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy/getRetryWaitTime(error:attempt:)":{"kind":"symbol","type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)","abstract":[{"text":"Returns whether we should retry and how long we should wait before retrying","type":"text"}],"role":"symbol","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"getRetryWaitTime"},{"text":"(","kind":"text"},{"text":"error","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","kind":"typeIdentifier","text":"Error"},{"kind":"text","text":", "},{"kind":"externalParam","text":"attempt"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:Si","text":"Int","kind":"typeIdentifier"},{"text":") -> ","kind":"text"},{"text":"RetryStatus","kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore11RetryStatusO"},{"text":"?","kind":"text"}],"required":true,"url":"\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)","title":"getRetryWaitTime(error:attempt:)"},"doc://codes.soto.soto-core/documentation/SotoCore":{"type":"topic","abstract":[{"type":"text","text":"The core framework for Soto, a Swift SDK for AWS"}],"kind":"symbol","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","url":"\/documentation\/sotocore","title":"SotoCore","role":"collection"},"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy":{"kind":"symbol","type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy","abstract":[{"text":"Protocol for Retry strategy. Has function returning amount of time before the next retry after an HTTP error","type":"text"}],"role":"symbol","fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"RetryPolicy","kind":"identifier"}],"navigatorTitle":[{"text":"RetryPolicy","kind":"identifier"}],"url":"\/documentation\/sotocore\/retrypolicy","title":"RetryPolicy"}}}